File: app/src/main/java/com/shentu/wallpaper/mvp/ui/adapter/CategoryListAdapter.java
package com.shentu.wallpaper.mvp.ui.adapter;

import android.widget.ImageView;

import androidx.annotation.Nullable;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.shentu.wallpaper.R;
import com.shentu.wallpaper.app.GlideArms;
import com.shentu.wallpaper.model.entity.Wallpaper;

import java.util.List;

public class CategoryListAdapter extends BaseQuickAdapter<Wallpaper, BaseViewHolder> {
    public CategoryListAdapter(@Nullable List<Wallpaper> data , int categoryId) {
        super(R.layout.item_rv_category_list, data);
    }

    @Override
    protected void convert(BaseViewHolder helper, Wallpaper item) {
        GlideArms.with(mContext)
                .load(item.getUrl())
                .placeholder(R.drawable.default_cover_vertical)
                .into((ImageView) helper.getView(R.id.iv_small));
    }
}

File: app/src/main/java/com/shentu/wallpaper/mvp/ui/adapter/CategoryListAdapter.kt
package com.shentu.wallpaper.mvp.ui.adapter

import android.view.View
import android.widget.ImageView
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import com.shentu.wallpaper.R
import com.shentu.wallpaper.app.GlideArms
import com.shentu.wallpaper.app.utils.HkUtils
import com.shentu.wallpaper.model.entity.Wallpaper

class CategoryListAdapter(data: List<Wallpaper>) : BaseQuickAdapter<Wallpaper
        , BaseViewHolder>(R.layout.item_rv_category_list, data) {
    override fun convert(helper: BaseViewHolder, item: Wallpaper) {
        GlideArms.with(mContext)
                .load(HkUtils.instance.get2x2Image(item.url))
                .placeholder(R.drawable.default_cover_vertical)
                .into((helper.getView<View>(R.id.iv_small) as ImageView))
    }
}